### PR TITLE
Updates to support Eclipse 2024-06

### PR DESCRIPTION
**Pull request by mezarin**

### CODE REVIEW

File: bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/debug/LibertySourcePathComputer.java [hunk 1]
```diff
@@ -132,12 +140,40 @@ private List<IProject> getProjectDependencies(Project project) throws CoreExcept
...
+                    // is downloaded and used by the connector. Gradle 8.1.1 does not support Java 21, 
+                    // which causes runtime issues (Unsupported class file major version 65).
+                    // As a workaround, specify a Java 21 compatible Gradle version that the tooling
+                    // can use (i.e. 8.4+). Note that since it is preferable to use the default version 
+                    // provided by the tooling API, setting the version can be revised at a later time.
+                    Throwable rootCause = Utils.findRootCause(e);
+                    if (rootCause != null && rootCause instanceof IllegalArgumentException) {
+                         String message = rootCause.getMessage();
+                         
+                         if (message != null && message.contains("Unsupported class file major version 65")) {
```
[the remark this answers is not in the document]
Comment (mezarin) [reply bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/debug/LibertySourcePathComputer.java:168]: The message is not something we produce, but something a message from the exception that we check. I know this is not optimal of preferred.

File: releng/target-platform-3Q-2024/target-platform-3Q2024.target [hunk 2]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
[the remark this answers is not in the document]
Comment (mezarin) [reply releng/target-platform-3Q-2024/target-platform-3Q2024.target]: Thanks for catching that. I updated it.

**Comment by scottkurz** [comment]
Noting this issue is held up by https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/3236 and https://github.com/google/gson/issues/2725.    

The likely resolution will be to wait for the next release of JDT LS with a likely target of the end of August, since it is already fixed in the main branch, rather than a more clever resolution of the issues.

Note a workaround is to delete each of these from the plugins pool dir: 
* com.google.gson_2.11.0
* org.eclipse.tips.json_0.3.400.* 

and then restart Eclipse with `-clean`.

**Review by scottkurz — APPROVED** [review]
LGTM

**Pull request merged**